### ISSUE TITLE
initramfs splash screen for: osk-sdl crashed

### ISSUE THREAD

**Issue by ollieparanoid** [postmarketOS/pmbootstrap#768]
After #476 is merged, this would be helpful. I think I had this situation twice, and it's hard to understand what happens, when you just have a black screen.
The message could display a short URL to a wiki page with information on how to get the log from the initramfs and open a bug report.

**Comment by craftyguy** [comment]
Great idea. The return value of osk-sdl could be checked to determine if it quit because the user unlocked it or pressed Esc, or if it quit because it crashed. In the former case, the current implementation is OK. For the latter case (crash), that would be a good time to show a splash screen or maybe cat log contents to display since there's no good way to get at the actual file itself (right?)

**Comment by ollieparanoid** [comment]
You can get to the file with the usb-shell hook.

`cat` to the display won't work in most cases, because we don't have the tty enabled for most Android devices (on `lg-mako` that made the boot take some extra **minutes**).

But we could `cat` it to stdout anyway, which will put it into the pmOS log file unless you specified in the kernel parameters, that the output should not be redirected. Then it would be possible to read it from serial or really on the display (e.g. in qemu).

**Comment by craftyguy** [comment]
So the message on the splash would have to be something like: 

> osk-sdl is having a bad day. to retrieve the log file, install pmOS with the usb-shell hook (insert wiki link here?) then do this, this, and this to log in, retrieve it, and post it so craftyguy can fix it!

**Comment by ollieparanoid** [comment]
Yeah, but we could put all that in the wiki article to save space:
> then do this, this, and this to log in, retrieve it, and post it so craftyguy can fix it!

**Comment by montvid** [comment]
Yes, it seems something crashes I can see a black screen but no pmos logo after unlocking with osk. I just reboot the device and it works after that.

**Comment by craftyguy** [comment]
@montvid

File a new ticket in the osk-sdl repo if you are experiencing a
problem, don't overload unrelated tickets :)

**Comment by montvid** [comment]
Oh, i thought you were talking about the same thing. Osk crashing. Oh well.
